crates/rpc/src/handlers/validator.rs

```diff
@@ -173,22 +200,84 @@ pub async fn get_validators_from_state(
         ));
     }
 
-    Ok(with_status(
-        BeaconResponse::json(validators_data),
-        StatusCode::OK,
-    ))
+    Ok(HttpResponse::Ok().json(BeaconResponse::new(validators_data)))
 }
 
+#[post("/beacon/states/{state_id}/validators")]
 pub async fn post_validators_from_state(
-    state_id: ID,
-    request: ValidatorsPostRequest,
-    db: ReamDB,
-) -> Result<impl Reply, Rejection> {
-    let id_query = IdQuery { id: request.ids };
+    db: Data<ReamDB>,
+    state_id: Path<ID>,
+    request: Json<ValidatorsPostRequest>,
+    status_query: Json<StatusQuery>,
+) -> Result<impl Responder, ApiError> {
+    let id_query = IdQuery {
+        id: request.ids.clone(),
+    };
 
     let status_query = StatusQuery {
-        status: request.status,
+        status: request.status.clone(),
```

Comment:
can you update this code to avoid the clone

```
let ValidatorsPostRequest { ids, status } = request.into_inner();

let status_query = StatusQuery { status };
```

You can remove `id_query` and just use the value directly below